astropy/units/utils.py

```diff
@@ -248,8 +252,10 @@ def resolve_fractions(a, b):
     This ensures that any operation involving a Fraction will use
     rational arithmetic and preserve precision.
     """
-    a_is_fraction = isinstance(a, Fraction)
```

Comment:
`isinstance(a, Fraction)` is awfully slow for the case that `a` is not a fraction. It might be possible to improve this further...

Reply:
Can you add a comment to this effect? Something like "checking common cases int and float first because isinstance(a, Fraction) is slow". I can foresee coming back to this code in some time and wanting to "clean it up" :)